src/plone/restapi/services/discussion/conversation.py

```diff
@@ -0,0 +1,203 @@
+# -*- coding: utf-8 -*-
+from AccessControl import getSecurityManager
+from Acquisition import aq_inner
+from plone.app.discussion.browser.comment import EditCommentForm
+from plone.app.discussion.browser.comments import CommentForm
+from plone.app.discussion.interfaces import IConversation
+from plone.app.discussion.interfaces import IDiscussionSettings
+from plone.registry.interfaces import IRegistry
+from plone.restapi.deserializer import json_body
+from plone.restapi.interfaces import ISerializeToJson
+from plone.restapi.services import Service
+from pkg_resources import get_distribution
+from zExceptions import BadRequest
+from zExceptions import Unauthorized
+from zope.component import getMultiAdapter
+from zope.component import queryUtility
+from zope.interface import alsoProvides
+from zope.interface import implementer
+from zope.publisher.interfaces import IPublishTraverse
+
+from datetime import datetime
+import plone.protect.interfaces
+
+PLONE4 = get_distribution('Products.CMFPlone').version < '5'
+
+
+class CommentsGet(Service):
+
+    def reply(self):
+        conversation = IConversation(self.context)
+        serializer = getMultiAdapter(
+            (conversation, self.request),
+            ISerializeToJson
+        )
+        return serializer()
+
+
+@implementer(IPublishTraverse)
+class CommentsAdd(Service):
+    comment_id = None
+
+    def publishTraverse(self, request, name):
+        if name:
+            self.comment_id = long(name)
+            request['form.widgets.in_reply_to'] = name
+        return self
+
+    def reply(self):
+        # Disable CSRF protection
+        if 'IDisableCSRFProtection' in dir(plone.protect.interfaces):
+            alsoProvides(self.request,
+                         plone.protect.interfaces.IDisableCSRFProtection)
+
+        conversation = IConversation(self.context)
+        if self.comment_id and self.comment_id not in conversation.keys():
+            self.request.response.setStatus(404)
+            return
+
+        # Fake request data
+        body = json_body(self.request)
+        for key, value in body.items():
+            self.request.form['form.widgets.' + key] = value
+
+        form = CommentForm(self.context, self.request)
+        form.update()
+
+        action = form.actions['comment']
+        data, errors = form.extractData()
+        if errors:
+            raise BadRequest({'errors': [err.error for err in errors]})
+
+        form.handleComment(form=form, action=action)
+
+        self.request.response.setStatus(204)
+        if 'location' in self.request.response.headers:
```

Comment:
@jaroel the server should set the location header to the URL of the newly created comment to allow the client to display the new comment.